src/counts.jl

```diff
@@ -369,18 +380,23 @@ function addcounts_radixsort!(cm::Dict{T}, x::AbstractArray{T}) where T
 end
 
 # fall-back for `x` an iterator
-function addcounts_radixsort!(cm::Dict{T}, x) where T 
-    sx = sort!(collect(x), alg = RadixSort)
+function addcounts_radixsort!(cm::Dict{T}, x) where T
+    sx = sort!(vec(collect(x)), alg = RadixSort)
     return _addcounts_radix_sort_loop!(cm, sx)
 end
 
 function addcounts!(cm::Dict{T}, x::AbstractArray{T}, wv::AbstractVector{W}) where {T,W<:Real}
-    n = length(x)
-    length(wv) == n || throw(DimensionMismatch())
+    # add wv weighted counts of integers from x to cm
+
+    length(x) == length(wv) ||
+        throw(DimensionMismatch("x and wv must have the same length, got $(length(x)) and $(length(wv))"))
+
+    xv = vec(x) # discard shape because weights() discards shape
```

Comment:
Thinking about this again, maybe we should deprecate this behavior by adding a separate method like this:
```julia
@deprecate addcounts!(cm::Dict{T}, x::AbstractArray{T}, wv::AbstractVector{W}) where {T,W<:Real} = addcounts!(cm, vec(x), wv)
```
And restrict the signature of this method to `x::AbstractVector{T}`.